whatshap/cli/split.py

```diff
@@ -224,22 +246,45 @@ def _fastq_binary_iterator(fastq_file):
         yield record.name, len(record.sequence), (str(record) + "\n").encode("utf-8")
 
 
-def check_haplotag_list_information(haplotag_list, exit_stack):
+def _gfa_file_iterator(gfa_file):
```

Comment:
this looks like it will break easily. afaik, the GFA format specifies the following line types:
comment/header/segment/link/containment/path - all of those should be handled explicitly, and any other non-empty line should raise an explicit error to make debugging easier.
Also, GFA files do not need to contain sequence information, so this way of computing the node length may not give the desired result.

Reply:
I added all possible gfa line records to the function.

Reply:
I realize that it would be "good practice" to add a small test file and write a test for the GFA parsing (at least I don't see that you changed any tests with this pull request?)